design/one-pager-stacks-security-isolation.md

```diff
@@ -56,6 +56,33 @@ From the [Kubernetes documentation](https://kubernetes.io/docs/tasks/access-kube
 Therefore, any time a CRD is created, it will also be at the cluster scope, even if instances of the CRD are scoped to a namespace.
 This has caused confusion in the past, so it's important to clear that up here.
 
+#### Discoverability of Scope
+
+Because CRDs are cluster scoped, but our permission model for stacks is both namespaced and global we will
+ need to label CRDs to indicate which are available in a given context. As the stack-installer processes a
+ cluster or namespace stack install it needs to add labels indicating that it's crossplane environment scoped or
+ namespace scoped to a given namespace.
```

Comment:
```
$ kubectl get -o json crd clusterstackinstalls.stacks.crossplane.io|jq .spec.scope
"Namespaced"
```

CRDs already encode this data as `.spec.scope`. Is the idea that we want to duplicate it as a label because we can't list filtered by `.spec.scope` (it's not a supported field selector)? That may be worthwhile, especially if we expect folks to want to list CRDs by scope using `kubectl`, but if we expect this list to be aggregated by some other UX it may be worth simply having the frontend list all CRDs and filter them manually.

Reply:
> A general theme; I'm not sure we should leak the Upbound internal concept of an "environment" into this Crossplane design document. Crossplane doesn't currently (as far as I know) have the concept of an environment.

This was pre-existing in this doc in master so I think it leaked into UbC :) If we want to remove, I guess I'll defer to you guys.

Reply:
I think an important feature here @negz is the ability to know **which** namespaces a CRD is "applicable" for, or in other words, which namespaces a stack has been installed into and therefore which namespaces a CRD has truly been intended to be used in.  e.g., the following query that gets all CRDs installed from stacks that are applicable to the `foo` namespace is not something you can do with the built in `.spec.scope` that CRDs have:

```
kubectl get crds --selector crossplane.io/scope=namespace,namespace.crossplane.io/foo=true
```

Reply:
I raised a similar concern here, https://github.com/crossplaneio/crossplane/pull/1008#discussion_r342233345

Yes, we can't filter the API fetched CRDs by `scope` (`kubectl` or otherwise) without some secondary filter.  Labels permit indexing and filtering for the Crossplane/Stack CRDs.